src/main/java/domain/score/Score.java

```diff
@@ -0,0 +1,45 @@
+package domain.score;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.WeakHashMap;
+
+public class Score {
+    private final double value;
+    private static final Map<Double, Score> CACHE = new WeakHashMap<>();
+
+    public static final Score ZERO = Score.of(0);
+
+    private Score(double value) {
+        this.value = value;
+    }
+
+    public static Score of(double value) {
+        return CACHE.computeIfAbsent(value, Score::new);
+    }
+
+    public Score add(Score score) {
+        return Score.of(this.value + score.value);
+    }
+
+    public Score half() {
+        return new Score(value / 2);
+    }
+
+    public double getScore() {
+        return value;
+    }
```

Comment:
getter는 필드명을 써주는 것이 맞습니다. 

```suggestion
    public double getValue() {
        return value;
    }
```